include/zephyr/bluetooth/conn.h

```diff
@@ -1828,6 +1828,20 @@ int bt_conn_auth_pairing_confirm(struct bt_conn *conn);
  */
 int bt_conn_auth_pincode_entry(struct bt_conn *conn, const char *pin);
 
+/** Converts a SMP error to string.
+ *
+ * The error codes are described in the Bluetooth Core specification,
+ * Vol 3, Part H, Section 3.5.5.
+ *
+ * The Security Manager Protocol documentation found in Vol 4, Part H,
+ * describes when the different error codes are used.
+ *
+ * See also the defined BT_SMP_ERR_* macros.
+ *
+ * @return The string representation of the SMP error code.
+ */
+const char *bt_smp_err_to_str(uint8_t smp_err);
```

Comment:
Should the error codes be moved to a public place? They are currently only defined in an internal header file, which actually makes this function useless :) 

Or did you mean to use this for the `enum bt_security_err` which is the public representation of the SMP error codes?

Reply:
I thought the same - the SMP error codes should be added to a public header. We should also add an API to convert `enum bt_security_err` to string

Reply:
But is the SMP error codes ever provided to the application? 
If not, then there's no point in providing a public function for a set of error values that never reach the public API. 

Perhaps the SMP converting function should be internal and just expose a public function for the `bt_security_err`.

Reply:
I agree. We should not add this API as a public API